herddb-core/src/main/java/herddb/model/Table.java

```diff
@@ -455,6 +452,21 @@ public Table build() {
                     tablespace, auto_increment, maxSerialPosition);
         }
 
+        private static boolean validatePrimaryKeyDataType(int type) {
+            switch(type) {
+                case ColumnTypes.NOTNULL_INTEGER:
+                case ColumnTypes.INTEGER:
```

Comment:
please add a comment to explain that we are allowing nullable data types in order to keep compatibility with pre-0.9 databases and clients